### PR TITLE
Prettier pretty printing

### PR DESCRIPTION
**Pull request by privat**
Some bugfixes, improvements and workaround for nitprettty.

The two options `--no-inline` and `--line-width` allow to control some rules about line-breaks.
This can be useful especially for people that want more relaxed rules.

~~~sh
$ ./nitpretty --check ../lib | wc -l
224
$ ./nitpretty --check ../lib --line-width 0 --no-inline | wc -l
188
~~~

So 36 more files are identical to their pretty printing with those relaxed rules.

For information, the tool crashed on some files in lib with the version before the PR.

Note that a lot of more work is still needed, one can see very strange and inconsistent results if the line width is forced to 1.

### CODE REVIEW

**Comment by Morriar** [comment]
+1!

**Comment by jpages** [comment]
+1

File: src/pretty.nit [hunk 1]
```diff
@@ -1269,10 +1291,11 @@ redef class AIfExpr
...
-				while not v.current_token isa TKwelse do
-					v.consume v.current_token.text
-				end
+			v.consume_comments
+
+			# FIXME: for some unknown reasons, has_else can be true even if
+			# there is no `else` keyword but a `end` instead.
+			if has_else(v) and v.current_token isa TKwelse then
 
 				v.indent -= 1
```
Comment (xymus) [inline src/pretty.nit]: The indent on the line manipulating indents is weird... I've heard of nitpretty that could clean this up for you.

**Comment by xymus** [comment]
+1

File: src/pretty.nit (hunk 1 above)
Comment (privat) [inline src/pretty.nit]: I do not see the weirdness: statements in a `if` are indented with a `\t`. Even the vim indenter agrees on this one.

**Pull request merged**